### PR TITLE
Add setup to run example with cookie authentication locally

### LINKED ISSUE #6096: Incorrect Instructions for with-cookie-auth example
# Examples bug report

## Example name
with-cookie-auth

## Describe the bug
As a developer evaluating this example, I'm not able to run it locally.

The instructions state to install and run using:

```
npm install
npm run dev
```

However, these commands do not work, because there is no `package.json` in the directory https://github.com/zeit/next.js/tree/canary/examples/with-cookie-auth

## To Reproduce

Download the example [or clone the repo](https://github.com/zeit/next.js):

```bash
curl https://codeload.github.com/zeit/next.js/tar.gz/canary | tar -xz --strip=2 next.js-canary/examples/with-cookie-auth
cd with-cookie-auth
```

Install it and run:

```bash
npm install
npm run dev
```

## Expected behavior
The example should start.

## Screenshots
image

## System information
 - OS: Ubuntu
 - Version of Next.js: [e.g. 6.0.2]

## Additional context

Additionally, when navigating to the `api` directory, the command `npm run dev` (which is defined in `package.json`) does not run, yielding the error:

image

The biggest problems with this example:
 * The instructions are not correct.
 * I was ultimately not able to start the API, as did not know the right commands to use.

### PR DESCRIPTION
**Pull request by juanbzz**
Fixes #6096 

- Includes an `index.js` file in the api folder to import the `login` and `profile` microservices and serve them as an API locally.
- Includes a custom `server.js` for the Next.js app to run locally and proxy the auth requests to the server API.
- Extracts the request logic from the `auth.js` file and keep concerns separated.
- Adds explanation of how everything is setup in the `README.md`

The goal for this PR is to make it easier to run the example locally while keeping the [monorepo](https://zeit.co/examples/monorepo/) file structure.

### CODE REVIEW

File: examples/with-cookie-auth/README.md [hunk 1]
```diff
@@ -21,13 +22,33 @@ curl https://codeload.github.com/zeit/next.js/tar.gz/canary | tar -xz --strip=2
 cd with-cookie-auth
 ```
 
-Install it and run:
+### Run locally
+
+The repository is setup as a [monorepo](https://zeit.co/examples/monorepo/) so you can deploy it easily running `now` inside the project folder. However, you can't run it the same way locally (yet).
```
Comment (curran) [inline examples/with-cookie-auth/README.md]: Great explanation!

File: examples/with-cookie-auth/README.md [hunk 2]
```diff
@@ -21,13 +22,33 @@ curl https://codeload.github.com/zeit/next.js/tar.gz/canary | tar -xz --strip=2
...
+
+These files make it easier to run the application locally and aren't needed for production:
```
Comment (curran) [inline examples/with-cookie-auth/README.md]: Maybe good to add a caveat here "aren't needed for production if you are using Now to deploy".

File: examples/with-cookie-auth/README.md [hunk 3]
```diff
@@ -40,8 +61,8 @@ In this example, we authenticate users and store a token in a cookie. The exampl
 
 This example is backend agnostic and uses [isomorphic-unfetch](https://www.npmjs.com/package/isomorphic-unfetch) to do the API calls on the client and the server.
 
-The repo includes a minimal passwordless backend built with [Micro](https://www.npmjs.com/package/micro) and it logs the user in with a GitHub username and saves the user id from the API call as token. 
+The repo includes a minimal passwordless backend built with [Micro](https://www.npmjs.com/package/micro) that logs the user in with a GitHub username and saves the user id from the API call as token.
 
 Session is syncronized across tabs. If you logout your session gets logged out on all the windows as well. We use the HOC `withAuthSync` for this.
 
-The helper function `auth` helps to retrieve the token across pages and redirects the user if no token was found.
+The helper function `auth` helps to retrieve the token across pages and redirects the user if not token was found.
```
Comment (curran) [inline examples/with-cookie-auth/README.md]: The original text here "if no token was found" is more correct in terms of English grammar than the new version "if not token was found". I would suggest to revert it back.

**Review by curran — APPROVED** [review]
Excellent to see the missing pieces come together! Thanks a ton for this PR.

**Review by dav-is — APPROVED** [review]

**Pull request merged**